plugin/pkg/admission/resourcequota/admission_test.go

```diff
@@ -860,6 +860,124 @@ func TestAdmitBelowTerminatingQuotaLimit(t *testing.T) {
 	}
 }
 
+// TestAdmitBelowTerminatingQuotaLimitWhenPodScopeUpdated ensures that terminating pods are charged to the right quota.
+// It creates a terminating and non-terminating quota, and changes an existing pod to be terminating.
+// It ensures that the terminating quota is incremented, and the non-terminating quota is not.
```

Comment:
Add the explanation of why we do this (always overcount if something something later causes the request to be rejected so you cannot reduce quota with requests that aren't completed).

Reply:
added.